R/lockfile.R

```diff
@@ -174,17 +174,30 @@ renv_lockfile_create <- function(project,
 
   lockfile <- renv_lockfile_init(project)
 
-  renv_lockfile_records(lockfile) <-
-
-    renv_snapshot_libpaths(libpaths = libpaths,
-                           project  = project) %>%
-
-    renv_snapshot_filter(project  = project,
-                         type     = type,
-                         packages = packages,
-                         exclude  = exclude) %>%
+  # Repeat with restarts so that renv_snapshot_filter() can install packages
+  # if requested by the user
+  repeat {
+    withRestarts(
+      {
+        records <- renv_snapshot_libpaths(libpaths = libpaths, project = project)
+        records <- renv_snapshot_filter(
+          project  = project,
+          records  = records,
+          type     = type,
+          packages = packages,
+          exclude  = exclude
+        )
+        # skipped if recomputeRecords restart is used
+        break
+      },
+      recomputeRecords = function() {
+        renv_dynamic_reset()
```

Comment:
I wonder if we should also set some value to ensure we don't prompt the user again, just in case? (To avoid potentially recursively prompting the user.)

This shouldn't happen if `install()` works correctly, but it feels like a good idea to have a safeguard given the structure of the code here.

Reply:
Yeah, maybe the `repeat` is just the wrong structure all together. Let me think about it.